### PR TITLE
Add In-page Error for Unauthorized Access to Question Authoring

### PR DESCRIPTION
**Pull request by noce2**
## Purpose

- This replaces the use of window.alert style message to warn users when they attempt to navigate to the authoring page without appropriate permissions.

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type

What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

- Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

- Test the code

Automated tests
```sh
npm test
npm run e2e

```

Manual test
- Navigate to `/practice/create` without logging in or as a `consumer`

## What to Check

Verify that the following are valid

- `npm test` 
```sh
Executed 69 of 72 (skipped 3) SUCCESS (2.334 secs / 1.971 secs)
TOTAL: 69 SUCCESS
```

- `npm run e2e` 
```sh
[1] Executed 12 of 12 specs SUCCESS in 10 secs
```

- Manual test: Instead of the authoring feature area, a message with a no-access logo should appear with a message and 2 buttons indicating possible actions for the user.

### CODE REVIEW

**Review by aakano684 — APPROVED** [review]
LGTM. Only thing to maybe improve readability is some of the tests have the same message strings in multiple places and canbe distracting when trying to figure out what the test is doing. So referencing those strings in another way might be a good idea.

File: src/app/components/practice/question-authoring-page/question-authoring-page.component.spec.ts [hunk 1]
```diff
@@ -58,16 +67,28 @@ describe('QuestionAuthoringPageComponent', () => {
...
-  it('should create', () => {
-    expect(component).toBeTruthy();
+  it('should create and not show the unauthorized error message if the user has permissions', () => {
+    const page = new QuestionAuthoringTestPage(fixture);
+    page.fixture.detectChanges();
+    expect(page.componentInstanceUnderTest).toBeTruthy();
+
+    expect(page.primaryFeatureArea).toBeTruthy('the primary feature area could not be seen');
+    expect(page.unauthorisedFeatureAccessMessage).toBeNull('the unauthorised feature access message could be seen');
   });
```
Comment (noce2cicd) [inline src/app/components/practice/question-authoring-page/question-authoring-page.component.spec.ts:80]: The warning on assertion fail can cause confusion for the reader. These will be removed as part of #55 .

File: src/app/components/practice/question-authoring-page/question-authoring-page.component.spec.ts [hunk 2]
```diff
@@ -58,16 +67,28 @@ describe('QuestionAuthoringPageComponent', () => {
...
 
+  it('should display only the error message if the user does not have permissions', async(() => {
+    // Given
+    questionAuthoringGuardStub.doesUserHavePermissions.returns(of(false));
+    const page = new QuestionAuthoringTestPage(TestBed.createComponent(QuestionAuthoringPageComponent));
+    page.fixture.detectChanges();
+
+    expect(page.unauthorisedFeatureAccessMessage).toBeTruthy('the unauthorised feature access message could not be seen');
```
Comment (noce2cicd) [inline src/app/components/practice/question-authoring-page/question-authoring-page.component.spec.ts:87]:
```suggestion
    expect(page.unauthorisedFeatureAccessMessage).toBeTruthy();
```

To be fixed in #55

File: src/app/components/practice/question-authoring-page/question-authoring-page.component.spec.ts [hunk 3]
```diff
@@ -58,16 +67,28 @@ describe('QuestionAuthoringPageComponent', () => {
...
+    expect((page.unauthorisedFeatureAccessMessageText.nativeElement as HTMLElement).innerText).toBe('You need to be a contributor to create questions.');
+    expect(page.primaryFeatureArea).toBeNull('the primary feature area could be seen');
```
Comment (noce2cicd) [inline src/app/components/practice/question-authoring-page/question-authoring-page.component.spec.ts:89]:
```suggestion
    expect(page.primaryFeatureArea).toBeNull();
```

To be fixed in #55

**Review by noce2cicd — APPROVED** [review]
Reiterating @aakano684's comments and approving as he did so already.

**Pull request merged**